### DISCUSSION TITLE
How to validate the appium:forceSimulatorSoftwareKeyboardPresence capability with simulator device

### DISCUSSION THREAD

**Discussion by svermagit** [appium/appium#16079] [Q&A] [answered]
Could you please let me know , how can I validate "appium:forceSimulatorSoftwareKeyboardPresence" capability while running a tests on simulator? Is anything there that can be seen on simulator device during test run.

**Comment by KazuCocoa** [comment, marked as answer]
When the capability is true, you can see software keyboard when a text field is selected. Then, your keyboard type on your mac also should not reflect to the simulator input.

**Reply by svermagit** [reply]
But the software keyboard on selection of text field gets visible even the capability is set as false. Could you please suggest if there any way I can see the specific impact on simulator device or if it make any change in iOS settings app related to keyboard.

**Reply by KazuCocoa** [reply]
set to `false` does nothing as:

> Appium respects preset simulator software/hardware keyboard preference when this value is false

in the description. https://github.com/appium/appium-xcuitest-driver#simulator
It means if your simulator preference enabled software keyboard, it appears.

https://github.com/appium/WebDriverAgent/blob/master/WebDriverAgentLib/Utilities/FBConfiguration.m#L299-L310 is the code when it sets to `true`.
